cmake/macros/CompositeSources.cmake

```diff
@@ -96,6 +127,7 @@ function(composite_sources target sources_var)
 
           # Reset for the next composite file.
```

Comment:
This block probably needs to be moved up one if level, because flushing out the composites can now happen due to the extension changing, so the previous assumption that `if(num_composite_sources GREATER 1)` was only false at the very end of compositing is no longer true. If `composite_sources` isn't cleared out, the file will just sit around and be added to the next composite file generated.

A way to reproduce the bug I'm suspecting would be to try compositing 2 .cxx files, 1 .mm file, and 2 .c files, in that order. Unless I'm missing something, the .mm file will end up sitting around in `composite_sources` and get added to the top of the generated .c file. (The expected outcome is the .cxx files and .c files are composited while the .mm file is left bare)